### PR TITLE
[WIP DO NOT MERGE] nRF52 driver

### PR DESCRIPTION
**Pull request by mlaz**
This is a first version of the driver, it still needs work.
This driver uses the PWM high level API defined on https://github.com/apache/mynewt-core/pull/448

### CODE REVIEW

File: hw/mcu/nordic_sdk/pkg.yml [hunk 1]
```diff
@@ -51,9 +51,9 @@ pkg.src_dirs:
     - "src/ext/nRF5_SDK_11.0.0_89a8197/components/drivers_nrf/"
     - "src/ext/nRF5_SDK_11.0.0_89a8197/components/libraries/fifo/"
     - "src/ext/nRF5_SDK_11.0.0_89a8197/components/libraries/util/"
-
+    #- "src/ext/nRF5_SDK_11.0.0_89a8197/components/libraries/pwm/"
```
Comment (mkiiskila) [inline hw/mcu/nordic_sdk/pkg.yml]: Does this need to be conditionally included in src_dirs?
I assume conditional include like
`pkg.src_dirs.PWM:`
`   - "src/ext/nRF5_SDK_11.0.0_89a8197/components/libraries/pwm/"`
should work. If not, we should add that to newt tool.

File: hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+/* Nordic headers */
+#include <bsp.h>
+#include <nrf.h>
+#include <nrf_pwm.h>
+#include <nrf_drv_pwm.h>
+#include <nrf_drv_clock.h>
+/* #include <bsp/cmsis_nvic.h> */
+#include <app_timer.h>
+
+//#include <app_error.h>
```
Comment (mkiiskila) [inline hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: Minor nit. We'd prefer /* */ style comments over //. I believe this is listed in coding style doc.

**Review by mkiiskila — APPROVED** [review]
This looks good. Once the main api is merged, we can merge this one right after.

File: hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c (hunk 2 above)
Comment (mlaz) [reply hw/drivers/pwm/pwm_nrf52/src/pwm_nrf52.c]: This doesn't need to be here. It is not used. Deleted.

File: hw/mcu/nordic_sdk/pkg.yml (hunk 1 above)
Comment (mlaz) [reply hw/mcu/nordic_sdk/pkg.yml]: This doesn't need to be here. It is not used. Deleted.

**Pull request merged**